src/sys/unix.rs

```diff
@@ -207,70 +211,56 @@ impl_debug!(
     libc::AF_INET6,
     libc::AF_UNIX,
     #[cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux"))]
+    #[cfg_attr(
+        docsrs,
+        doc(cfg(any(target_os = "android", target_os = "fuchsia", target_os = "linux")))
+    )]
     libc::AF_PACKET,
     #[cfg(any(target_os = "android", target_os = "linux"))]
+    #[cfg_attr(docsrs, doc(cfg(any(target_os = "android", target_os = "linux"))))]
     libc::AF_VSOCK,
     libc::AF_UNSPEC, // = 0.
 );
 
 /// Unix only API.
+#[cfg(any(
+    target_os = "android",
+    target_os = "dragonfly",
+    target_os = "freebsd",
+    target_os = "fuchsia",
+    target_os = "illumos",
+    target_os = "linux",
+    target_os = "netbsd",
+    target_os = "openbsd"
+))]
+#[cfg_attr(
+    docsrs,
+    doc(cfg(any(
+        target_os = "android",
+        target_os = "dragonfly",
+        target_os = "freebsd",
+        target_os = "fuchsia",
+        target_os = "illumos",
+        target_os = "linux",
+        target_os = "netbsd",
+        target_os = "openbsd"
+    )))
+)]
 impl Type {
     /// Set `SOCK_NONBLOCK` on the `Type`.
-    ///
-    /// # Notes
-    ///
-    /// This function is only available on Android, DragonFlyBSD, Fuchsia,
-    /// FreeBSD, Linux, NetBSD and OpenBSD.
-    #[cfg(all(
-        feature = "all",
-        any(
-            target_os = "android",
-            target_os = "dragonfly",
-            target_os = "freebsd",
-            target_os = "fuchsia",
-            target_os = "illumos",
-            target_os = "linux",
-            target_os = "netbsd",
-            target_os = "openbsd"
-        )
-    ))]
+    #[cfg(feature = "all")]
```

Comment:
Please revert to the old `cfg` as Apple (macOS, iOS) don't have `SOCK_NONBLOCK`, same for the `cfg_attr` below.

Reply:
This won't be enabled on Apple because the above `#[cfg]` that applies to this entire impl block excludes Apple.

I just factored out the common "android, dragonfly, freebsd, fuchsia, illumos, linux, netbsd, openbsd" cfg that previously applied to every item in the impl block individually to the impl block itself, to avoid having to write out the 23-line long attributes out each time.

Reply:
I know it's easier for now, but when we add a new function/constant for macOS we would have to revert this again. I know it's overly verbose but I think I prefer the `cfg` to be complete for a given function without having to consider the `cfg` on the impl blocks, I think it makes reading it easier.